st2exporter/st2exporter/worker.py

```diff
@@ -13,118 +13,65 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import Queue
-
 import eventlet
+
 from kombu import Connection
-from oslo_config import cfg
 
 from st2common import log as logging
-from st2common.constants.action import (LIVEACTION_STATUS_SUCCEEDED, LIVEACTION_STATUS_FAILED,
-                                        LIVEACTION_STATUS_CANCELED)
-from st2common.models.api.execution import ActionExecutionAPI
-from st2common.models.db.execution import ActionExecutionDB
-from st2common.persistence.execution import ActionExecution
-from st2common.persistence.marker import DumperMarker
-from st2common.transport import consumers, execution, publishers
+from st2common.transport import execution, publishers
 from st2common.transport import utils as transport_utils
-from st2common.util import isotime
-from st2exporter.exporter.dumper import Dumper
+from st2exporter.consumers.execution import ExecutionsExporter
 
-__all__ = [
-    'ExecutionsExporter'
-]
-
-COMPLETION_STATUSES = [LIVEACTION_STATUS_SUCCEEDED, LIVEACTION_STATUS_FAILED,
-                       LIVEACTION_STATUS_CANCELED]
-LOG = logging.getLogger(__name__)
 
 EXPORTER_WORK_Q = execution.get_queue(
     'st2.exporter.work', routing_key=publishers.UPDATE_RK)
+LOG = logging.getLogger(__name__)
+
 
+class Exporter(object):
+    def __init__(self):
+        self._model_exporters = []
 
-class ExecutionsExporter(consumers.MessageHandler):
-    message_type = ActionExecutionDB
+        self._executions_exporter = None
+        with Connection(transport_utils.get_messaging_urls()) as conn:
+            self._executions_exporter = ExecutionsExporter(conn, [EXPORTER_WORK_Q])
 
-    def __init__(self, connection, queues):
-        super(ExecutionsExporter, self).__init__(connection, queues)
-        self.pending_executions = Queue.Queue()
-        self._dumper = Dumper(queue=self.pending_executions,
-                              export_dir=cfg.CONF.exporter.dump_dir)
-        self._consumer_thread = None
+        self._model_exporters.append(self._executions_exporter)
+        LOG.info('Available model exporters %s.', self._model_exporters)
+        self._model_exporters_pool = eventlet.GreenPool(len(self._model_exporters))
+        self._stopped = False
 
     def start(self, wait=False):
-        LOG.info('Bootstrapping executions from db...')
-        try:
-            self._bootstrap()
-        except:
-            LOG.exception('Unable to bootstrap executions from db. Aborting.')
-            raise
-        self._consumer_thread = eventlet.spawn(super(ExecutionsExporter, self).start, wait=True)
-        self._dumper.start()
+        LOG.info('Starting model exporters...')
+        for model_exporter in self._model_exporters:
+            while self._model_exporters_pool.free() <= 0 and not self._stopped:
```

Comment:
Hm, if I'm reading this correctly (which I might not be since the diff is a bit hard to read), this condition makes no sense.

Exporters are long running "processes" (right?) so in case `free()` returns `< 0` this means we simply miss-configured the pool size at the beginning since the pool should be static (we are not dynamically adding or removing exporters, size == number of exporter workers).

The pool size would only change if an existing exporter dies or is shut down, right?